core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -162,6 +162,7 @@ class Partition(val topic: String,
    */
   def makeLeader(controllerId: Int,
                  partitionStateInfo: PartitionStateInfo, correlationId: Int): Boolean = {
+    var leaderHWIncremented = false
```

Comment:
This is not a general "let's mutate everything" argument. I too favor functional programming, but I also like having return values that make sense. Anyway, I probably won't be drinking the scala kool-aid any time soon, but I could get on board with rewriting Kafka in Clojure. :)